metrics-exporter-log/src/lib.rs

```diff
@@ -57,46 +55,26 @@ where
     }
 
     /// Run this exporter, logging output only once.
-    pub fn turn(&self)
+    pub fn turn(&mut self)
     where
-        C: SnapshotProvider,
-        C::SnapshotError: Error,
+        C: Record,
     {
-        match self.controller.get_snapshot() {
-            Ok(snapshot) => {
-                let mut recorder = self.recorder.clone();
-                snapshot.record(&mut recorder);
-                let output = recorder.into();
-                log!(self.level, "{}", output);
-            }
-            Err(e) => log!(Level::Error, "failed to get snapshot: {}", e),
-        }
+        self.controller.record(&mut self.recorder);
+        let output = self.recorder.export();
+        log!(self.level, "{}", output);
+        self.recorder.flush();
     }
 
     /// Converts this exporter into a future which logs output on the given interval.
-    pub fn into_future(self, interval: Duration) -> impl Future<Item = (), Error = ()>
+    pub fn run_async(mut self, interval: Duration) -> impl Future<Item = (), Error = ()>
```

Comment:
I don't think think this should be renamed.  It's not actually running anything for the user, it's giving them the future that _they_ need to run.

Reply:
From what in understand of the [`convention`] , a `into_` convertion can't have any parameters. Thats why I renamed it. But I agree my new name sucks.

[`convention`]: https://rust-lang-nursery.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
For instance https://docs.rs/futures/0.1.28/futures/future/trait.IntoFuture.html.

Reply:
How about `interval_async`?

Reply:
I'm thinking of putting the `Duration` directly directly into the type instead. I think its an acceptable tradeoff since its unlikely that a user would create a `LogExporter` and exclusively use `turn`.

Reply:
That way we would get the same API than the http-exporter.

Reply:
Yeah, I'm fine with making the interval be a field on the exporter. 👍